explainaboard/loaders/loader.py

```diff
@@ -87,78 +128,66 @@ def user_defined_metadata_configs(self) -> dict:
+
+                custom_feature_configs: dict[str, CustomFeature] = {}
+                if custom_features:  # add custom features to output file loader fields
+                    custom_feature_configs = {
+                        name: CustomFeature.from_dict(name, dikt)
+                        for name, dikt in raw_data[
+                            "user_defined_features_configs"
+                        ].items()
+                    }
+                    fields: list[FileLoaderField] = []
+                    for feature in custom_feature_configs.values():
+                        fields.append(
+                            # dtype is set to None because custom feature configs
+                            # doesn't use the same set of dtypes as FileLoader
+                            # (this is not enforced anywhere)
+                            FileLoaderField(feature.name, feature.name, None, False)
+                        )
+                    self._output_file_loader.add_fields(fields)
+                return custom_feature_configs, metadata
+            return {}, {}
+        else:
+            raise Exception(
+                "_parse_user_defined_fields can only be called with a JSON system "
+                + "output file"
             )
-            if isinstance(raw_data, dict) and raw_data.get(
-                "user_defined_metadata_configs"
-            ):
-                self._data = json.dumps(raw_data["predictions"])
-                self._source = Source.in_memory
-                return raw_data["user_defined_metadata_configs"]
-        return {}
-
-    def load(self) -> Iterable[dict]:
-        file_loader = self.file_loaders[self._file_type]
-        return file_loader.load(self._data, self._source)
-
-
-# loader_registry is a global variable, storing all basic loading functions
-_loader_registry: dict[TaskType, type[Loader]] = {}
-
-
-def get_loader(
-    task: TaskType | str,
-    data: str,
-    source: Source | None = None,
-    file_type: FileType | str | None = None,
-) -> Loader:
-    task_cast: TaskType = TaskType(task)
-    file_type_cast: FileType | None = (
-        FileType(file_type) if file_type is not None else None
-    )
-    return _loader_registry[task_cast](data, source, file_type_cast)
 
-
-def register_loader(task_type: TaskType):
-    """
-    a register for different data loaders, for example
-    For example, `@register_loader(TaskType.text_classification)`
-    """
-
-    def register_loader_fn(cls):
-        _loader_registry[task_type] = cls
-        return cls
-
-    return register_loader_fn
+    def load(self) -> list[dict]:
+        dataset_loaded_data = self._dataset_file_loader.load(
+            self._dataset_data, self._dataset_source
+        )
+        output_loaded_data = self._output_file_loader.load(
+            self._output_data, self._output_source
+        )
+        if len(dataset_loaded_data) != len(output_loaded_data):
+            raise Exception(
```

Comment:
Maybe `ValueError`? I try to avoid just `Exception`.

Reply:
Actually, this validation isn't relevant anymore. `self._user_defined_features_configs` and `self._user_defined_metadata_configs` are always of dict types (this is enforced by type hints) so I have removed this.